src/common/system.h

```diff
@@ -3,8 +3,8 @@
 // Distributed under the MIT software license, see the accompanying
 // file COPYING or http://www.opensource.org/licenses/mit-license.php.
 
-#ifndef BITCOIN_UTIL_SYSTEM_H
-#define BITCOIN_UTIL_SYSTEM_H
+#ifndef BITCOIN_COMMON_SYSTEM_H
+#define BITCOIN_COMMON_SYSTEM_H
```

Comment:
In commit "refactor: Move system from util to common library" (0a7b3e5058b58c448bbc473ffb347910d781c3bd)

I think it makes sense to keep `util::insert` and `util::AnyPtr` functions in `util/`, not `common/` as these are header-only template functions with no dependencies, and there's no particular reason they shouldn't be used by kernel code even if they aren't used right now. In terms of functionality they are also more related to other helper functions like `util/overloaded.h` `util/strings.h` `util/time.h` `util/types.h` and `util/vector.h` than to anything in `common/`

I think ideally the `util::insert` function might move to `util/insert.h` and the `util::AnyPtr` function might move to `util/any.h`, but you could also just leave them in `util/system.h` if that is easier.

Reply:
Mmh, I was originally thinking of moving `util::insert` to `wallet/walletutil.h` and `util::AnyPtr` to `rpc/util.h` in a follow up after this pull request. I think your suggestion is better though. Moving around these kind of headers depending on which part of the code needs them seems silly. I'll patch this then and will apply your other suggestions.